packages/next/server/dev/next-dev-server.ts

```diff
@@ -324,36 +317,42 @@ export default class DevServer extends Server {
             pageName = pageName.replace(/\/index$/, '') || '/'
           }
 
-          if (regexMiddleware.test(fileName)) {
-            routedMiddleware.push(
-              `/${relative(this.pagesDir, fileName).replace(/\\+/g, '/')}`
-                .replace(/^\/+/g, '/')
-                .replace(regexMiddleware, '/')
-            )
+          if (pageName === '/_middleware') {
+            routedMiddleware.push(`/`)
             continue
           }
 
-          invalidatePageRuntimeCache(fileName, safeTime)
-          const pageRuntimeConfig = (
-            await getPageStaticInfo(fileName, this.nextConfig)
-          ).runtime
-          const isEdgeRuntime = pageRuntimeConfig === 'edge'
-
-          if (
-            isEdgeRuntime &&
-            !(isReservedPage(pageName) || isCustomErrorPage(pageName))
-          ) {
-            routedMiddleware.push(pageName)
-            ssrMiddleware.add(pageName)
+          /**
+           * If there is a middleware that is not declared in the root we will
+           * warn without adding it so it doesn't make its way into the system.
+           */
+          if (/[\\\\/]_middleware$/.test(pageName)) {
+            Log.warn(
```

Comment:
```suggestion
            Log.error(
```
Since this is an error in a production build we probably want to treat it as an error here as well